src/_pytest/fixtures.py

```diff
@@ -1539,45 +1529,38 @@ def _getautousenames(self, nodeid: str) -> Iterator[str]:
 
     def getfixtureclosure(
         self,
-        fixturenames: Tuple[str, ...],
         parentnode: nodes.Node,
+        initialnames: Tuple[str],
+        arg2fixturedefs: Dict[str, Sequence[FixtureDef[Any]]],
         ignore_args: Sequence[str] = (),
-    ) -> Tuple[Tuple[str, ...], List[str], Dict[str, Sequence[FixtureDef[Any]]]]:
+    ) -> List[str]:
```

Comment:
I would prefer to keep returning a brand new `arg2fixturedefs`, and let the caller merge it with other dict if they want, rather than passing in a dict and fill it inside. This makes it clear what's input/output.

Reply:
But the second fixture closure computation needs `arg2fixturedefs` as input, otherwise it should compute it again which is expensive according to docstring:
```python
# ........ we also populate arg2fixturedefs mapping
# for the args missing therein so that the caller can reuse it and does
# not have to re-discover fixturedefs again for each fixturename
# (discovering matching fixtures for a given name/node is expensive).
```

Reply:
How about this?
```python
def getfixtureclosure(
        self,
        parentnode: nodes.Node,
        initialnames: Tuple[str, ...],
        arg2fixturedefs: Union[Dict[str, Sequence[FixtureDef[Any]]], None],
        ignore_args: Sequence[str] = (),
    ) -> Tuple[List[str], Dict[str, Sequence[FixtureDef[Any]]]]:
```
If `arg2fixturedefs` is none, we create one, populate it and return it. Otherwise we just use it and return it.